enterprise/coderd/provisionerdaemons.go

```diff
@@ -101,14 +102,21 @@ func (p *provisionerDaemonAuth) authorize(r *http.Request, orgID uuid.UUID, tags
 		}
```

Comment:
Is it possible that `provAuth` is true and `apiKeyMW` is too? Should we just return here? If the user is not authorized, but using an api key, why continue?

Reply:
The current behavior of the client is it provides the api key and the psk / provisioner key all as headers. If an authenticated user doesn't have permission it falls back to see if a psk / provisioner key is provided instead.

Reply:
Feels like we should not allow that because it could confuse and org an user provisioner :thinking:

Reply:
Yeah we already do though, and idk if I'd want to change that behavior for existing users of the authenticated cli that use a psk today.

Reply:
Is it a lot of code to change the behavior if they use a key vs a PSK? Fix the behavior for the new method, and keep the old?

Reply:
Good call, because the PSK behavior will be phased out either way. I've updated it to now prevent both user and provisioner key auth at the same time.